### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by zjhmale**
refactor for #70, @rikvdkleij please review.

### CODE REVIEW

File: src/main/scala/intellij/haskell/action/CreateHaskellFileAction.scala [hunk 1]
```diff
@@ -58,48 +58,54 @@ class CreateHaskellFileAction extends CreateFileFromTemplateAction(CreateHaskell
...
-    val name = mkdirs.newName
-    val dir = mkdirs.directory
-    val project = dir.getProject
-
-    val nameWithmodulePrefix = if (pathItems == null || invalidPathItems(pathItems) || pathItems.isEmpty) {
-      name
-    } else {
-      pathItems.mkString(".") + "." + name
-    }
+    val pathItems = HaskellFileUtil.getPathFromSourceRoot(originDir.getProject, originDir.getVirtualFile)
```
Comment (zjhmale) [inline src/main/scala/intellij/haskell/action/CreateHaskellFileAction.scala]: @rikvdkleij can not use `originDir.getContainingFile` here, or there will be an NPE.

Comment (rikvdkleij) [inline src/main/scala/intellij/haskell/action/CreateHaskellFileAction.scala]: I see, weird, with files it's safer to use getContainingFile. The PsiDirectoryImpl just returns null...

**Comment by rikvdkleij** [comment]
@zjhmale  I did small refactor of your code after merging.

**Comment by zjhmale** [comment]
@rikvdkleij Great! I saw the commit [here](https://github.com/rikvdkleij/intellij-haskell/commit/eaf7a863d553451b1c3bf7d1ba543ad41ad671ba), and `foreach` in Scala is just the same as `fmap` in Haskell right?

**Comment by rikvdkleij** [comment]
@zjhmale No, `foreach` is for side effects.

**Comment by zjhmale** [comment]
@rikvdkleij ok, got it.

**Comment by rikvdkleij** [comment]
@zjhmale Create Haskell file in test directory does not work anymore. It's related to this PR. I will fix it.

**Comment by rikvdkleij** [comment]
I reverted all your code because I do not see easy fix and I have to continue with developing my Haskell application.

**Pull request merged**